release.sh

```diff
@@ -98,95 +109,119 @@ if [[ "$CONFIRM" != "y" ]]; then
+elif [[ ! $PUBLISH ]]; then
+    # Make a release branch
+    git checkout -b "v$VERSION-release"
+
+    # Edit package.json to the right version, as it's inlined (see
+    # http://stackoverflow.com/a/22084103 for why we need the .bak file to make
+    # this mac & linux compatible)
+    sed -i.bak -E 's|"version": "[^"]+",|"version": "'$VERSION'",|' package.json
+    rm -f package.json.bak
 
-# Make the commit and tag, and push them.
-git add package.json README.md contrib/*/README.md dist/README.md \
-    docs/*.md website/pages/index.html website/versioned_docs/ \
-    website/versioned_sidebars/ website/versions.json
-git commit -n -m "v$VERSION"
-git diff --stat --exit-code # check for uncommitted changes
-git tag -a "v$VERSION" -m "v$VERSION"
-git push origin "v$VERSION"
+    # Build generated files
+    yarn build
+
+    if [ ! -z "$NEXT_VERSION" ]; then
+        # Edit package.json to the next version
+        sed -i.bak -E 's|"version": "[^"]+",|"version": "'$NEXT_VERSION'-pre",|' package.json
+        rm -f package.json.bak
+        git add package.json
+    fi
+
+    # Edit docs to use CSS from CDN
+    grep -l '{@stylesheet: katex.min.css}' docs/*.md | xargs sed -i.bak \
+        's|{@stylesheet: katex.min.css}|<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@./dist/katex.min.css" integrity="sha384-katex.min.css" crossorigin="anonymous"/>|'
+
+    # Update the version number in CDN URLs included in the README and the documentation,
+    # and regenerate the Subresource Integrity hash for these files.
+    node update-sri.js "${VERSION}" README.md contrib/*/README.md \
+        docs/*.md docs/*.md.bak website/pages/index.html
+
+    # Generate a new version of the docs
+    pushd website
+    yarn run version "${VERSION}"
+    popd
+
+    # Restore docs to use local built CSS
+    for file in docs/*.md.bak; do
+        mv -f "$file" "${file%.bak}"
+    done
 
-# Update npm (cdnjs update automatically)
-yarn publish --new-version "${VERSION}"
+    # Make the commit and push
+    git add package.json README.md contrib/*/README.md \
+        docs website/pages/index.html website/versioned_docs/ \
+        website/versioned_sidebars/ website/versions.json
+    if [[ -z "$NEXT_VERSION" ]]; then
+        git commit -n -m "Release v$VERSION"
+    else
+        git commit -n -m "Release v$VERSION" -m "Bump $BRANCH to v$NEXT_VERSION-pre"
+    fi
+    git diff --stat --exit-code # check for uncommitted changes
+    git push -u origin "v$VERSION-release"
 
-# Go back to original branch to bump
-git checkout "$BRANCH"
+    echo ""
+    echo "The automatic parts are done!"
+    echo "Now create a pull request against master from 'v$VERSION-release'"
+    echo "Visit https://github.com/Khan/KaTeX/pulls to open a pull request."
+    echo "After it gets merged, run './release.sh -p $VERSION'!"
+    echo "Note that if KaTeX source code is changed after running this script,"
+    echo "you have to run the release script again."
```

Comment:
If you run `./release.sh 0.6.3 0.6.4` will it push a new commit to the `v0.6.3-release` branch created by the first run?

Reply:
@kevinbarabash Yes, and the next version is not used for the second run, so it can be omitted.